### PR TITLE
roachtest: bugfix testeng grafana link missing cluster name

### LINKED ISSUE #107894: roachtest: grafana link does not have cluster name
Currently (as of 07/28/23), in a roachtest failure on 23.2 (like #107862), the generated Grafana link in the "Help" section redirects you to cockroachlabs.com - instead of testeng Grafana. 

Looking at the issue linked above as an example, we see the following vanity link for Grafana: https://go.crdb.dev/p/roachfana//1690711427218/1690713270780

Instead, we should be seeing the following:
https://go.crdb.dev/p/roachfana/teamcity-11114747-1690696061-26-n5cpu4/1690711427218/1690713270780

The cluster name seems to be empty - `clusterName` in `generateHelpCommand`.

Jira issue: CRDB-30247

### PR DESCRIPTION
**Pull request by annrpom**
Previously, the testeng grafana link generated after a roachtest failure directed the user to cockroachlabs.com due to a missing cluster name from the link. This patch should fix this issue by getting the cluster name from a `*githubIssues.cluster.name` instead of the `clusterName` from roachtest/cluster.go.

Fixes: #107894

Release note (bug fix): The link to testeng grafana that is generated on a roachtest failure should now properly direct to grafana.

### CODE REVIEW

**Comment by annrpom** [comment]
tl;dr: my guess is that the cause of this is related to the fact that clusters are wiped before and after each test runs: https://github.com/cockroachdb/cockroach/pull/22795 (?) 
not entirely sure if i should've even used this `clusterName` in the first place~ 

I noticed that using `clusterName` defined in roachtest/cluster.go (implemented in the previously linked PR) will return the empty string due to the reason above; so, I believe using `g.cluster.name` as the argument to `generateHelpCommand` in `createPostRequest` will resolve this? I did need to put a guard in `generateHelpCommand` that kinda goes with the logic of "if cluster creation fails, there will be no Grafana link"
- this was to avoid a nil ref error in case we try to get the name of a cluster when cluster is nil + a way to avoid a faulty Grafana link that takes you to cockroachlabs.com
  - (is this intuitive or will people be confused as to why the Grafana link is missing sometimes?)

I can for sure commit some better tests (i.e. putting some echotests in TestCreatePostRequest) since I realized that testing how generateHelpCommand works inside of createPostRequest helped me through this process.

**Comment by renatolabs** [comment]
> I believe using `g.cluster.name` as the argument to generateHelpCommand in createPostRequest will resolve this?

I believe so too.

> this was to avoid a nil ref error in case we try to get the name of a cluster when cluster is nil + a way to avoid a faulty Grafana link that takes you to cockroachlabs.com 

Makes sense to me.

> (is this intuitive or will people be confused as to why the Grafana link is missing sometimes?)

I'd say it's good enough. In practice, the `cluster` reference should only be `nil` when cluster creation fails and those errors are aggregated in issues like #105298. For every actual test failure, the Grafana link should be there as per your current logic (and its absence should continue be treated as a bug).

File: pkg/cmd/roachtest/github.go [hunk 1]
```diff
@@ -60,10 +60,12 @@ func generateHelpCommand(
 			"How To Investigate (internal)",
 			"https://cockroachlabs.atlassian.net/l/c/SSSBr8c7",
 		)(renderer)
-		issues.HelpCommandAsLink(
-			"Grafana",
-			fmt.Sprintf("https://go.crdb.dev/p/roachfana/%s/%d/%d", clusterName, start.UnixMilli(), end.UnixMilli()),
-		)(renderer)
+		if clusterName != "" {
```
Comment (renatolabs) [inline pkg/cmd/roachtest/github.go:63]: Might be worth adding a note that an empty `clusterName` here should correspond to a cluster creation failure.

**Review by renatolabs — APPROVED** [review]
Thank you for fixing!

File: pkg/cmd/roachtest/github.go (hunk 1 above)
Comment (annrpom) [reply pkg/cmd/roachtest/github.go:63]: sounds good! will add that

**Comment by annrpom** [comment]
TFTR! ('-')7

bors r=renatolabs
